File: 9-1.py
#!/usr/bin/env python

inputs = [int(row) for row in open('9.input').readlines()]

lookback = 25

row = lookback
while row < len(inputs):
    is_sum = False
    for x in range(row - lookback - 1, row):
        if is_sum:
            break

        for y in range(x, row):
            if inputs[x] + inputs[y] == inputs[row]:
                is_sum = True

    if not is_sum:
        print(f"{inputs[row]}")
        break

    row += 1

File: 9-1.rb
#!/usr/bin/env ruby

inputs = File.read('9.input').lines.map(&:to_i)

lookback = 25

row = lookback
while row < inputs.size
  is_sum = false
  (row - lookback - 1).upto(row - 1) do |x|
    break if is_sum

    x.upto(row - 1) do |y|
      is_sum = true if inputs[x] + inputs[y] == inputs[row]
    end
  end
  unless is_sum
    print "#{inputs[row]}\n"
    return
  end
  row += 1
end
